[InstagramBridge] Truncate long titles and use full text as content

lorenzos opened the pull request:
Instagram posts' text can be very long used as feed items' title. This PR truncates titles over 120 characters, and uses the full text as items' content if truncated.

Possible changes I can make on this PR if you prefer:

* Always put the full text in items' content, also if it was short enough to not be truncated in items' title.
* Use a better truncation function that preserves words' integrity.

lorenzos committed:
[InstagramBridge] Truncate long titles and use full text as content

lorenzos committed:
[InstagramBridge] Fixed code style

logmanoriginal commented:
Thanks for the PR!

>Always put the full text in items' content, also if it was short enough to not be truncated in items' title.

It makes sense to be consistent for all items in a feed in my opinion.

>Use a better truncation function that preserves words' integrity.

Does something like this work for you?
https://github.com/RSS-Bridge/rss-bridge/blob/5305c405f65826a0cf3bd0050cc10bdfdc4da19c/bridges/FacebookBridge.php#L664-L665

Let me know if you want to update the PR, I have no objections merging it like it is.

lorenzos commented:
I had something different for the truncation on words, but I guess it's ok if already used in the Facebook bridge. Do you want to use a 64 chars limit as in Facebook, or keep my proposed 120?

> Let me know if you want to update the PR

Sure, I will.

logmanoriginal commented:
>Do you want to use a 64 chars limit as in Facebook, or keep my proposed 120?

If 120 works for you, it's fine by me. in the end it depends on how your feed reader displays the titles. Since I don't use Instagram, this is no issue for me :)

lorenzos committed:
[InstagramBridge] Better title truncation, text context in all posts preserving new lines

lorenzos committed:
[InstagramBridge] Use only the first line of text content as title

lorenzos commented:
I think the PR is now ready to merge:

* I modified the title truncation as you suggested, using `wordwrap()`;
* I made that regardless of actual text content length, so when posts have explicit wraps only the first line goes in the title, this usually leaves hastags and crap aside, check [this](https://www.instagram.com/p/BqyCoZ-Fqcw/) for example;
* full text content is now added after the image in every item, and new lines are preserved using `nl2br()`.

logmanoriginal commented:
Merged. Thanks for keeping it updated :+1:
